querybook/webapp/components/DataDocQueryExecutions/DataDocQueryExecutions.tsx

```diff
@@ -24,136 +24,160 @@ interface IProps {
-                return null;
-            }
+            const handleQueryExecutionSelected = useCallback(
+                (qid: number) => {
+                    const newIndex = queryExecutions.findIndex(
+                        ({ id }) => qid === id
+                    );
+                    if (newIndex >= 0) {
+                        selectExecutionIndex(newIndex);
+                    }
+                },
+                [queryExecutions]
+            );
 
-            return (
-                <div className="execution-selector-section horizontal-space-between">
-                    <div className="flex-row">
-                        <QueryExecutionPicker
-                            queryExecutionId={currentExecution?.id}
-                            onSelection={handleQueryExecutionSelected}
-                            queryExecutions={queryExecutions}
-                        />
-                        <div className="ml8">
-                            <QueryExecutionDuration
+            const generateExecutionsPickerDOM = () => {
+                const currentExecution =
+                    queryExecutions?.[selectedExecutionIndex];
+                if (!currentExecution) {
+                    return null;
+                }
+
+                return (
+                    <div className="execution-selector-section horizontal-space-between">
+                        <div className="flex-row">
+                            <QueryExecutionPicker
+                                queryExecutionId={currentExecution?.id}
+                                onSelection={handleQueryExecutionSelected}
+                                queryExecutions={queryExecutions}
+                            />
+                            <div className="ml8">
+                                <QueryExecutionDuration
+                                    queryExecution={currentExecution}
+                                />
+                            </div>
+                        </div>
+                        <div className="execution-selector-bottom flex-row">
+                            <QueryExecutionBar
                                 queryExecution={currentExecution}
                             />
                         </div>
                     </div>
-                    <div className="execution-selector-bottom flex-row">
-                        <QueryExecutionBar queryExecution={currentExecution} />
-                    </div>
-                </div>
+                );
+            };
+
+            const selectedExecution = queryExecutions[selectedExecutionIndex];
+            const queryExecutionDOM = selectedExecution && (
+                <QueryExecution
+                    id={selectedExecution.id}
+                    docId={docId}
+                    key={selectedExecution.id}
+                    changeCellContext={changeCellContext}
+                    onSamplingInfoClick={onSamplingInfoClick}
```

Comment:
Only change in this file was passing 2 props here, rest at auto formatted changes